File: insertionSort.js

function insertionSort(array) {

  if ( array.length < 2 ) { return array; }

  for (var i = 0; i < array.length; i++) {
    var current = array[i];
    for (var j = i - 1; j >= 0 && array[j] > current; j--) {
      array[j+1] = array[j];
    }
    array[j+1] = current;
  }

  return array;

}

var testArray = [324,23,230,56,234,456,2,3,111,24]

console.log(insertionSort(testArray));

File: insertionSort.go
package main

import "fmt"

func insertionSort(list []int) []int {

	for i := 0; i < len(list); i++ {
		var j int
		value := list[i]
		for j = i - 1; j > -1 && list[j] > value; j-- {
			list[j+1] = list[j]
		}
		list[j+1] = value
	}

	return list
}

func main() {
	test := []int{49, 54, 98, 19, 70, 2, 23, 76, 26, 99, 6, 106, 11}
	fmt.Println(insertionSort(test))
}
